feat: pull_request_template

wesleyboar opened the pull request:
## Overview

Add a template for pull request descriptions.

## Changes

- **added** `pull_request_template.md`

## Testing / UI

https://github.com/DesignSafe-CI/DS-User-Guide/blob/e17c3417/.github/pull_request_template.md

## Notes

Same as is used on:

- https://github.com/TACC/Core-CMS
- https://github.com/TACC/Core-Styles
- https://github.com/TACC/Core-Portal
- https://github.com/TACC/Core-CMS-Custom
- https://github.com/TACC/TACC-Docs

wesleyboar commented on pyproject.toml:
```diff
@@ -2,6 +2,7 @@
 name = "ds-user-guide"
 version = "0.1.0"
 description = ""
+package-mode = false
```
This avoids a warning in newer version of Poetry, which user may install using "Local Development" → "via Docker", and is [accurate](https://python-poetry.org/docs/pyproject/#package-mode).

<details><summary>The Warning</summary>

> Installing the current project: ds-user-guide (0.1.0)
Warning: The current project could not be installed: No file/folder found for package ds-user-guide
If you do not want to install the current project use --no-root.
If you want to use Poetry only for dependency management but not for packaging, you can disable package mode by setting package-mode = false in your pyproject.toml file.
In a future version of Poetry this warning will become an error!

</details>

silviamazzoni replied:
poetry is lower case...

wesleyboar replied:
Unless one is pretentious enough to usurp a dictionary word as the name of their product, yes.

wesleyboar commented on Dockerfile:
```diff
@@ -5,7 +5,7 @@ LABEL maintainer="TACC-ACI-WMA <wma_prtl@tacc.utexas.edu>"
 
 # https://python-poetry.org/docs/configuration/#using-environment-variables
-ENV POETRY_VERSION=1.4.2 \
+ENV POETRY_VERSION=1.8.2 \
```
If user follows "Local Development" → "via Python", they will install the latest Python. Those instructions force us to keep Docker's version up to date to ensure compatibility. Alternatively, I update instructions to install Poetry at a specific version.

silviamazzoni replied:
I would add a comment that you may not need the latest python version. it worked for my 10.13, afterall.

wesleyboar replied:
Known supported versions are [>=3.10 and <3.12](https://github.com/DesignSafe-CI/DS-User-Guide/blob/fix/update-poetry-and-improve-instructions/pyproject.toml#L9). I've added note via a1f1c57.

silviamazzoni commented on README.md:
I would say:
Install dependencies (you need to do this only once) or something like this

looks great!

silviamazzoni replied:
also,
5. Open the website at the URL returned in the last of the lines echoed by the program, e.g.
    http://127.0.0.1:8000/user-guide/

wesleyboar replied:
Thanks. Adjusted via 97c172e.

silviamazzoni reviewed (approved):
thank you for taking the time to do this!

kks32 commented:
Looks good! @wesleyboar

wesleyboar committed:
feat: pull_request_template

Same as is used on:
- TACC/Core-CMS
- TACC/Core-Styles
- TACC/Core-Portal
- TACC/Core-CMS-Custom
- TACC/TACC-Docs
